solution.js

```diff
@@ -1,5 +1,184 @@
+    }
+    node.next = foundNode.next;
+    //takes the current node and drops the foundNode and connects to the next value
+  }
+  getFirst() {
+    return this.head;
+  }
+
+  getLast() {
+    if (this.head == null) {
+      return "Null list; no last element";
+    }
+    let currentNode = this.head;
+    while (currentNode.next !== null) {
+      currentNode = currentNode.next;
+    }
+    return currentNode;
+  }
+  search(key) {
+    if (this.head == null) {
+      return "No elements in list; key not found";
+    }
+    let currentNode = this.head;
+    while (currentNode) {
+      if (currentNode.data === key) {
+        return currentNode;
+      }
+      currentNode = currentNode.next;
+    }
+    return "key not found in list.";
+  }
+  getKth(k) {
+    let currentNode = this.head;
+    for (let i = 1; i < k; i++) {
+      if (currentNode) {
+        currentNode = currentNode.next;
+      }
+    }
+    if (currentNode) {
+      return currentNode;
+    } else {
+      return null;
+    }
+  }
+  // getKthToLast(k) {
+  //   if (k <= 1) {
+  //     return null; // Invalid value of k
+  //   }
+  
+  //   let slow = this.head;
+  //   let fast = this.head;
+  //   console.log(slow, fast)
+  //   // this.head.next = //one ahead of slow -- fast will hit last slow is one behind.
+  
+  //   // Move the fast pointer k positions ahead
+
+  //   for (let i = 0; i < k; i++) {
+  //     if (fast === null) {
+  //      throw new Error("The linked list has fewer than k nodes")  ; // throw an error if fast pointer is null
+  //     }
+  //     fast = fast.next;
+  //     // console.log(fast, fast.next);
+  //   }
+  
+  //   // Move both pointers simultaneously until the fast pointer reaches the end
+  //   while (fast !== null) { // the end.
+  //     // slow = slow.next;
+  //     //next after first
+  //     // fast = fast.next;//next after second ?
+  //   }
+  
+  //   return slow;
+  // }
+  getKthToLast(key) { // you used a key to get the last ok this makes sense
+    let currentNode = this.head; // used the current placeholder for the current
+    let size = this.size(); //this takes in the size of the list linked list ok
+
+    while (size - 1 > key) { // we want the current placeholder amount FROM LAST
+      // console.log(currentNode);
+      currentNode = currentNode.next; // reassign the current placeholder to the next to iterate
+      size--; //descending in the queue -- we queue fro the bottom up
+    };
+    return currentNode; // where and what
+  };
+  isEmpty(){
+    return (this.head == null) 
+  }
+
+  clear() {
+    this.head = null;
+  }
+
+  toArray() {
+    if (this.head == null) {
+      return [];
+    } else {
+      let thisNode = this.head;
+      const getArray = [];
+      while (thisNode) {
+        getArray.push(thisNode.data);
+        thisNode = thisNode.next;
+      }
+      return getArray;
+    }
+  }
+  containsDuplicates() {
+    const set = new Set();
+    let currentNode = this.head;
+  
```

Comment:
Using a set you could compare the length off the set to length of the linked list for shorter, cleaner code